### PR TITLE
Set WINVER

### LINKED ISSUE #4570: there is always a "import error" in a new win7 system: "ImportError: cannot import name 'get_all_providers'"
webwxgetmsgimg
Describe the bug
when I install onnxruntime in a new  windows 7 sp1 system, there is always a "import error" . I have already installed “vs 2019 runtime” （VC_redist.x64.exe）

Package           Version
----------------- -------
numpy             1.19.0
onnx              1.7.0
onnxruntime       1.4.0
pip               20.1.1
protobuf          3.12.2
setuptools        39.0.1
six               1.15.0
typing-extensions 3.7.4.2
wheel             0.34.2

C:\Users\zhjs>python
Python 3.6.6 (v3.6.6:4cf1f54eb7, Jun 27 2018, 03:37:03) [MSC v.1900 64 bit (AMD6
4)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import onnxruntime
C:\Users\zhjs\AppData\Local\Programs\Python\Python36\lib\site-packages\onnxrunti
me\capi\_pybind_state.py:14: UserWarning: Cannot load onnxruntime.capi. Error: '
DLL load failed: 找不到指定的程序。'.
  warnings.warn("Cannot load onnxruntime.capi. Error: '{0}'.".format(str(e)))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\zhjs\AppData\Local\Programs\Python\Python36\lib\site-packages\o
nnxruntime\__init__.py", line 13, in <module>
    from onnxruntime.capi._pybind_state import get_all_providers, get_available_
providers, get_device, set_seed, \
ImportError: cannot import name 'get_all_providers'
>>>

Urgency
High, our project is based on onnxruntime, and now it cannot run in a new win7 system，and we already installed “VC_redist.x64.exe”，we don‘t know what's missing？

System information

Win7:
pip insatll onnxruntime:
onnxruntime 1.4.0/1.3.0:
python 3.6.6
use cpu:
To Reproduce
 it cannot run in a new win7 system，and we already installed “VC_redist.x64.exe”，we don‘t know what's missing？

Expected behavior
no import error

Screenshots

error

### PR DESCRIPTION
**Pull request by tiagoshibata**
**Description**: Fixes #4570 . There's an internal email thread discussing this change.

### CODE REVIEW

File: cmake/CMakeLists.txt [hunk 1]
```diff
@@ -162,6 +162,7 @@ if(NOT WIN32)
     set(onnxruntime_ENABLE_INSTRUMENT OFF)
   endif()
 else()
+  add_compile_definitions(WINVER=0x0601)  # Support Windows 7 and newer
```
Comment (snnn) [inline cmake/CMakeLists.txt:165]: I think WINVER is controlled by CMAKE_SYSTEM_VERSION cmake variable.

Comment (snnn) [reply cmake/CMakeLists.txt:165]: I'll investigate more and reach you back later.

Comment (tiagoshibata) [reply cmake/CMakeLists.txt:165]: CMAKE_SYSTEM_VERSION points to the target SDK version. I'm pretty sure they are different things. Also, just searched through the CMake repo and there's no built-in variable for WINVER

Comment (tiagoshibata) [reply cmake/CMakeLists.txt:165]: And adding an external /D define or a define on top of headers are the way recommended by the MS docs: https://docs.microsoft.com/en-us/windows/win32/winprog/using-the-windows-headers#setting-winver-or-_win32_winnt

**Pull request merged**